docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
++
+[source,java]
+----
+package de.spinscale.javalin;
+
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.binder.jvm.JvmCompilationMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmHeapPressureMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
+import io.micrometer.core.instrument.binder.logging.Log4j2Metrics;
+import io.micrometer.core.instrument.binder.system.FileDescriptorMetrics;
+import io.micrometer.core.instrument.binder.system.ProcessorMetrics;
+import io.micrometer.core.instrument.binder.system.UptimeMetrics;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+
+public class Micrometer {
+
+    final PrometheusMeterRegistry registry = new PrometheusMeterRegistry(new PrometheusConfig() {
+        @Override
+        public String get(String key) {
+            return null;
+        }
+
+        @Override
+        public String prefix() {
+            return "javalin";
+        }
+    });
+
+    public Micrometer() {
+        Metrics.addRegistry(registry);
+        new JvmGcMetrics().bindTo(Metrics.globalRegistry);
+        new JvmHeapPressureMetrics().bindTo(Metrics.globalRegistry);
+        new JvmThreadMetrics().bindTo(Metrics.globalRegistry);
+        new JvmCompilationMetrics().bindTo(Metrics.globalRegistry);
+        new JvmMemoryMetrics().bindTo(Metrics.globalRegistry);
+        new Log4j2Metrics().bindTo(Metrics.globalRegistry);
+        new UptimeMetrics().bindTo(Metrics.globalRegistry);
+        new FileDescriptorMetrics().bindTo(Metrics.globalRegistry);
+        new ProcessorMetrics().bindTo(Metrics.globalRegistry);
+    }
+
+    public String scrape() {
+        return registry.scrape();
+    }
+}
+----
++
+. Restart your app and poll the metrics endpoint.
++
+[source,bash]
+----
+curl localhost:7000/metrics -u metrics:secret
+----
++
+This returns a line based response with one metric per line. This is the
+standard Prometheus format.
+
+[discrete]
+==== Install and configure {metricbeat}
+
+. Time to install and configure {metricbeat}.
+For details, see <<install-metricbeat,Install {metricbeat}>>.
++
+. Similar to the {filebeat} setup, run the initial set up of all the dashboards
+using the admin user, and then use an API key as well.
++
+[source,console]
+----
+POST /_security/api_key
+{
+  "name": "metricbeat_javalin-app",
+  "role_descriptors": {
+    "metricbeat_writer": {
+      "cluster": ["monitor", "read_ilm"],
+      "index": [
+        {
+          "names": ["metricbeat-*"],
+          "privileges": ["view_index_metadata", "create_doc"]
+        }
+      ]
+    }
+  }
+}
+----
++
+Don’t forget to do the initial setup like this.
++
+[source,bash]
+----
+./metricbeat setup -e -E 'cloud.id=${CLOUD_ID}' -E 'cloud.auth=elastic:YOUR_SUPER_SECRET_PASS'
```

Comment:
same comment about command syntax applies here. If we reuse our "widgets" for displaying commands, we'll need to add them for other commands. We could do this after this PR is merged to keep this review clean and move it along.